node/e2e-tests/test-service/tests/standalone.rs

```diff
@@ -130,3 +135,232 @@ async fn transaction_pool_priority_order_test() {
 	assert!(tx1.priority > tx2.priority);
 	assert!(tx2.priority > tx3.priority);
 }
+
+/// this testcase will take too long to run, test with command:
+/// cargo test --release --package test-service -- evm_fill_block_test --nocapture --include-ignored
+#[substrate_test_utils::test]
+#[ignore]
+async fn evm_fill_block_test() {
+	let mut builder = sc_cli::LoggerBuilder::new("");
+	builder.with_colors(true);
+	let _ = builder.init();
+
+	let para_id = ParaId::from(2000);
+	let tokio_handle = tokio::runtime::Handle::current();
+
+	let node = test_service::TestNodeBuilder::new(para_id, tokio_handle.clone(), Alice)
+		.with_seal_mode(SealMode::DevAuraSeal)
+		.enable_collator()
+		.build()
+		.await;
+
+	node.wait_for_blocks(1).await;
+
+	type Balances = pallet_balances::Pallet<node_runtime::Runtime>;
+
+	let acc = node.with_state(|| {
+		<node_runtime::Runtime as module_evm::Config>::AddressMapping::get_account_id(&H160::from(hex!(
+			"1000000000000000000000000000000000000001"
+		)))
+	});
+
+	let old_balance = node.with_state(|| Balances::free_balance(acc.clone()));
+
+	let target = H160::from(hex!("0000000000000000000100000000000000000000")); // ACA
+
+	// transfer
+	// to
+	// amount 100000000000
+	let input = hex! {"
+		a9059cbb
+		000000000000000000000000 1000000000000000000000000000000000000001
+		00000000000000000000000000000000 0000000000000000000000174876e800
+	"};
+
+	let functions = std::iter::repeat_with(|| {
+		node_runtime::Call::EVM(module_evm::Call::call {
+			target,
+			input: input.to_vec(),
+			value: 0,
+			gas_limit: 100_000,
+			storage_limit: 100_000,
+			access_list: vec![],
+		})
+	})
+	.take(1_000)
+	.collect();
+
+	frame_support::assert_ok!(node.submit_extrinsic_batch(functions, Some(Alice), 0).await);
+
```

Comment:
```
	let pending_tx = node.transaction_pool.status().ready as u128;
	assert_eq!(pending_tx, 1000);
	
	
	// wait for 1 blocks
	node.wait_for_blocks(1).await;
	
	let pending_tx = node.transaction_pool.status().ready as u128;
	assert_eq!(pending_tx, 1000 - 246);
```

One block can contain 246 ACA transfers.

What about creating a contract?
It seems that it can only contain 3 txs.

Reply:
https://github.com/AcalaNetwork/Acala/issues/1801#issuecomment-1046630602

Reply:
> ```
> 
> 	let pending_tx = node.transaction_pool.status().ready as u128;
> 
> 	assert_eq!(pending_tx, 1000);
> 
> 	
> 
> 	
> 
> 	// wait for 1 blocks
> 
> 	node.wait_for_blocks(1).await;
> 
> 	
> 
> 	let pending_tx = node.transaction_pool.status().ready as u128;
> 
> 	assert_eq!(pending_tx, 1000 - 246);
> 
> ```
> 
> 
> 
> One block can contain 246 ACA transfers.
> 
> 
> 
> What about creating a contract?
> 
> It seems that it can only contain 2 txs.
> 
> 	

I'm not testing create. That contract is deployed to test gas limit

Reply:
@zjb0807 added create fill block test

Reply:
```
+       let pending_tx = node.transaction_pool.status().ready as u128;
+       assert_eq!(pending_tx, 1000);
+       node.wait_for_blocks(2).await;
+       let pending_tx = node.transaction_pool.status().ready as u128;
+       assert_eq!(pending_tx, 1000 - 229);
```

Can you add the code?  We can track changes on the number.

Reply:
@zjb0807 these numbers will depend on weights, so a small weight change may require to update tx count as well and may not be convenient

Reply:
This is `#[ignore]`. Updates are only required for manual testing.

Reply:
@zjb0807 no they'll run. `make test-e2e` will include ignored